### PR TITLE
Add shellcheck and review the errors/warnings it raises.

### LINKED ISSUE #185: Introduce shellcheck
Please consider running shellcheck like in https://github.com/kward/shunit2/blob/master/.travis.yml#L24

### PR DESCRIPTION
**Pull request by drazisil**
## Purpose
What is the context of this pull request? Why is it being done?

Fixes #185 

Add shellcheck to test for bash errors https://www.shellcheck.net/

## Notable Changes
Are there any changes that need to be called out as particularly tricky or significant?

CI only change, so probably not

## Tests and Risks?
Is this covered by existing tests? New ones? If no, why not?

It's kinda a test itself, no additional tests needed

### CODE REVIEW

**Comment by szepeviktor** [comment]
It is `env -i PATH="$PATH" shellcheck codecov tests/test`

**Comment by drazisil** [comment]
This found a number of errors and I'm not confident to blindly change them. Will be slow going as I review.

File: codecov [hunk 1]
```diff
@@ -1300,8 +1300,8 @@ then
   fi
 fi
 
-upload_file=`mktemp /tmp/codecov.XXXXXX`
-adjustments_file=`mktemp /tmp/codecov.adjustments.XXXXXX`
+upload_file=$(mktemp /tmp/codecov.XXXXXX)
+adjustments_file=%(mktemp /tmp/codecov.adjustments.XXXXXX)
```
Comment (szepeviktor) [inline codecov]: @drazisil here we have a `%` !!

Comment (drazisil) [reply codecov]: Fixed.

**Comment by thomasrockhu** [comment]
Closing this out, because we did [it](https://github.com/codecov/codecov-bash/pull/313)!

**Pull request closed without merging**